server/src/tasks/wallet.js

```diff
@@ -100,7 +100,33 @@ const setWalletOwner = withAccount(async (account, { walletAddress, newOwner },
   return receipt
 })
 
+const createForeignWallet = withAccount(async (account, { userWallet, ens = '' }, job) => {
+  const { createContract, createMethod, send } = createNetwork('foreign', account)
+  const owner = userWallet.accountAddress
+  const walletFactory = createContract(WalletFactoryABI, userWallet.walletFactoryCurrentAddress)
+  const method = createMethod(walletFactory, 'createCounterfactualWallet', owner, Object.values(userWallet.walletModules), ens, userWallet.salt)
```

Comment:
this supposed to be original params not correct ones so we can be sure that the wallet will have the same address